### PR TITLE
Adding logic to set OMP_NUM_THREADS appropriately

### PR DESCRIPTION
**Pull request by etrain**
Addresses #198 

Tested locally on OSX in several combinations of situations, and verified that env variables were set as expected.

### CODE REVIEW

File: bin/run-pipeline.sh [hunk 1]
```diff
@@ -6,6 +6,32 @@ FWDIR="$(cd `dirname $0`; pwd)"
 CLASS=$1
 shift
 
+# Set OMP_NUM_THREADS on workers and driver to something appropriate.
+# This is due to OpenBLAS not handling large numbers of cores very well.
```
Comment (shivaram) [inline bin/run-pipeline.sh]: Lets add a link to the keystone issue here. I'll add my openblas commit investigation to the same issue

**Comment by shivaram** [comment]
Change looks fine. Can we test this on Linux on say a millennium machine before merging ?

File: bin/run-pipeline.sh (hunk 1 above)
Comment (etrain) [inline bin/run-pipeline.sh]: Will do.

**Comment by etrain** [comment]
Great - I will test on millennium this AM.

**Comment by etrain** [comment]
Apparrently `[[` and `-eq` don't behave as expected with strings. Once that was fixed everything seems to work fine on both linux and osx.

**Comment by shivaram** [comment]
Cool. LGTM

**Pull request merged**